### ISSUE TITLE
Can gatsby-plugin-feed be disabled?

### ISSUE THREAD

**Issue by danielclough** [styxlab/gatsby-theme-try-ghost#13]
I do not need an RSS feed at the moment and I am getting this error.
How can I disable gatsby-plugin-feed?

```
10:26:24 PM: error "gatsby-plugin-feed" threw an error while running the onPostBuild lifecycle:
10:26:24 PM: Cannot read property 'parent' of null
10:26:24 PM:   174 |             _options$feed = (0, _extends2.default)({}, options, {}, feed), setup = _options$feed.setup, locals = (0, _objectWithoutPropertiesLoose2.default)(_options$feed, ["setup"]);
10:26:24 PM:   175 |             serializer = feed.serialize && typeof feed.serialize === "function" ? feed.serialize : serialize;
10:26:24 PM: > 176 |             rssFeed = serializer(locals).reduce(function (merged, item) {
10:26:24 PM:       |                       ^
10:26:24 PM:   177 |               merged.item(item);
10:26:24 PM:   178 |               return merged;
10:26:24 PM:   179 |             }, new _rss.default(setup(locals)));
10:26:24 PM: 
10:26:24 PM:   TypeError: Cannot read property 'parent' of null
10:26:24 PM:   
10:26:24 PM:   - parse.js:69 Function.exports.update
10:26:24 PM:     [repo]/[cheerio]/lib/parse.js:69:26
10:26:24 PM:   
10:26:24 PM:   - parse.js:19 module.exports
10:26:24 PM:     [repo]/[cheerio]/lib/parse.js:19:11
10:26:24 PM:   
10:26:24 PM:   - static.js:27 Function.exports.load
10:26:24 PM:     [repo]/[cheerio]/lib/static.js:27:14
10:26:24 PM:   
10:26:24 PM:   - generate-feed.js:8 generateItem
10:26:24 PM:     [repo]/[gatsby-theme-try-ghost]/src/utils/rss/generate-feed.js:8:33
10:26:24 PM:   
10:26:24 PM:   - generate-feed.js:49 allGhostPost.edges.map.edge
10:26:24 PM:     [repo]/[gatsby-theme-try-ghost]/src/utils/rss/generate-feed.js:49:102
10:26:24 PM:   
10:26:24 PM:   - Array.map
10:26:24 PM:   
10:26:24 PM:   - generate-feed.js:49 serialize
10:26:24 PM:     [repo]/[gatsby-theme-try-ghost]/src/utils/rss/generate-feed.js:49:72
10:26:24 PM:   
10:26:24 PM:   - gatsby-node.js:176 _callee2$
10:26:24 PM:     [repo]/[gatsby-plugin-feed]/gatsby-node.js:176:23
10:26:24 PM:   
10:26:24 PM:   - next_tick.js:68 process._tickCallback
10:26:24 PM:     internal/process/next_tick.js:68:7
10:26:24 PM:   
10:26:24 PM: 
10:26:24 PM: not finished onPostBuild - 0.181s
10:26:24 PM: (sharp:1430): GLib-CRITICAL **: 05:26:24.522: g_hash_table_lookup: assertion 'hash_table != NULL' failed
10:26:24 PM: (sharp:1430): GLib-CRITICAL **: 05:26:24.612: g_hash_table_lookup: assertion 'hash_table != NULL' failed
10:26:24 PM: Skipping functions preparation step: no functions directory set
10:26:24 PM: Caching artifacts
10:26:24 PM: Started saving node modules
10:26:24 PM: Finished saving node modules
10:26:24 PM: Started saving yarn cache
10:26:24 PM: Finished saving yarn cache
10:26:24 PM: Started saving pip cache
10:26:24 PM: Finished saving pip cache
10:26:24 PM: Started saving emacs cask dependencies
10:26:25 PM: failed during stage 'building site': Build script returned non-zero exit code: 1
10:26:24 PM: Finished saving emacs cask dependencies
10:26:24 PM: Started saving maven dependencies
10:26:25 PM: Finished saving maven dependencies
10:26:25 PM: Shutting down logging, 7 messages pending
```

**Comment by styxlab** [comment]
Feeds can be disabled. Just go to the *subdirectory* `gatsby-theme-try-ghost` locate `gatsby-config.js` and comment out the line where feeds are generated:

```javascript
//gatsby-config.js
{
    resolve: `gatsby-plugin-feed`,
    options: {
        query: `
        {
            allGhostSettings {
                edges {
                    node {
                        title
                        description
                    }
                }
            }
        }
      `,
        feeds: [
            //generateRSSFeed(siteConfig),
        ],
    },
},
```

Based on the error message above it looks like one of your posts contains malformed `html` which leads to a feed generation failure. If you want us to look into this error further, please provide a content API key and a publicly available endpoint for debugging.

**Comment by danielclough** [comment]
Thank you.

You can use this to debug the problem.
"apiUrl": "https://cosplay.exposed",
"contentApiKey": "e6c796b2e7eeab8b025a1f4e59"

---

Is there a way to have the ghost admin on the same domain as the gatsby
site, or do I have to use subdomains?

**Comment by styxlab** [comment]
@danielclough Thanks for providing an API key. I was able to reproduce the issue with this. Problem was that your post with title "Street Fighter" does not contain any body text. This leaves the blog post field `post.html` empty and the RSS feed generation was not prepared to deal with empty posts.

I fixed the issue in the `master` branch of this repo and also published a new version `gatsby-theme-try-ghost@1.2.12` so people using the the Gatsby starter `gatsby-starter-try-ghost` do not run into this error.
